src/App.tsx

```diff
@@ -15,47 +16,131 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: number,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((currentGood: string, nextGood: string) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return currentGood.localeCompare(nextGood);
+
+      case SortType.LENGTH:
+        return currentGood.length - nextGood.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
```

Comment:
actually it would be great to separate the reverse logic, it will help with code maintenance in the future.
You should avoid accepting more than 2 args in the function if it is possible. That makes your code more complicated